file_sync_down/main.py

```diff
@@ -0,0 +1,53 @@
+import config, os, shutil, requests, binascii, zipfile, time, hashlib
+
+def download_single_file(filename: str):
+    response = requests.get(config.server_url + "/file?filename=" + filename)
+    assert response.status_code == 200
+
+    with open(config.sync_folder_location + "/" + filename, "wb+") as f:
+        f.write(response.content)
+
+def download_zip_file(files_to_download: list):
+    tmp_dir = "downloaded_zips/"
+    os.makedirs(tmp_dir, exist_ok=True)
+    comma_separated_list = ','.join(files_to_download)
+    response = requests.get(config.server_url + "/files?filenames=" + comma_separated_list)
+    assert response.status_code == 200
+
+    local_name = tmp_dir + binascii.b2a_hex(os.urandom(8)).decode("utf-8") + ".zip"
+
+    with open(local_name, "wb+") as f:
+        f.write(response.content)
+
+    return local_name
+
+# gets its parameter from the return value of download_zip_file
+def unzip_file(local_name: str):
+    with zipfile.ZipFile(local_name, 'r') as zip_ref:
+        zip_ref.extractall(config.sync_folder_location)
+
+def sync():
+    while True:
+        list_response = requests.get(config.server_url + "/list")
+        assert list_response.status_code == 200
+        all_files = list_response.json()["files"]
+        
+        already_downloaded_files = []
```

Comment:
Can make this a hashmap to speed up file comparison

Reply:
Certainly a potential optimization, but I feel like this is unlikely to be a bottleneck? Especially if the size of the list gets pruned by telling the server to "delete" anyway